.github/workflows/py-run-tests.yml

```diff
@@ -17,8 +19,16 @@ permissions:
   contents: read
 
 jobs:
-  test:
+  test_and_quality_check:
     runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: ["3.11","3.12","3.13"]
+    env:
+      PYTHON: "${{ matrix.python-version }}"
+    permissions:
+      contents: read  # checkout code only
```

Comment:
Adding Python versions to make sure we don't have breaking changes in different Python versions.

Renamed the workflow to future quality checks.